rig/machine_control/utils.py

```diff
@@ -0,0 +1,78 @@
+import six
+from ..machine import Cores, SDRAM
+
+
+def sdram_alloc_for_vertices(controller, placements, allocations,
+                             core_as_tag=True,
+                             sdram_resource=SDRAM, cores_resource=Cores):
+    """Allocate and return a file-like view of a region of SDRAM for each
+    vertex which uses SDRAM as a resource.
+
+    The tag assigned to each region of assigned SDRAM is the index of the
+    first core that each vertex is assigned.  For example::
+
+        placements = {vertex: (0, 5)}
+        allocations = {vertex: {Cores: slice(3, 6),
+                                SDRAM: slice(204, 304)}}
+        sdram_allocations = sdram_alloc_for_vertices(
+            controller, placements, allocations
+        )
+
+    Will allocate a 100-byte block of SDRAM for the vertex which is
+    allocated cores 3-5 on chip (0, 5).  The region of SDRAM will be tagged
+    `3` (because this is the index of the first core).
+
+    Parameters
+    ----------
+    controller : :py:class:`rig.machine_control.MachineController`
+        Controller to use to allocate the SDRAM.
+    placements : {vertex: (x, y), ...}
+        Mapping of vertices to the chips they have been placed on.  Same as
+        produced by placers.
+    allocations : {vertex: {resource: allocation, ...}, ...}
+        Mapping of vertices to the resources they have been allocated.  As
```

Comment:
Mapping of vertices to the resources they have been allocated.

A block of memory of the size specified by the `sdram_resource` (default:
:py:class:`~rig.machine.SDRAM`) resource will be allocated for each vertex. Note
that location of the supplied allocation is *not* used.

When `core_as_tag=True`, the tag allocated to allocations will be the ID of the
first core used by the vertex, otherwise the tag will be set to 0.